pybigquery/sqlalchemy_bigquery.py

```diff
@@ -12,6 +13,11 @@
 from sqlalchemy.engine.default import DefaultDialect, DefaultExecutionContext
 from sqlalchemy.engine.base import Engine
 from sqlalchemy.sql.schema import Column
+from sqlalchemy.sql import elements
+import re
+
+
+FIELD_LEGAL_CHARACTERS = re.compile('[^\w+r]+')
```

Comment:
should this variable be named FIELD_ILLEGAL_CHARACTERS as per the regex definition?
Also, why is this not sufficient `[^\w]+`?
Other criteria for legal field name is it must beginning with alphabet or underscore.
https://github.com/apache/incubator-superset/pull/5641/files

Reply:
You are right, thanks.

For the first character I am doing a check on line 79 and append underscore if it is invalid:

```python
if not name[0].isalpha() and name[0] != '_':
    name = "_" + name
```